include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -667,9 +667,17 @@ struct __result_and_scratch_storage : __result_and_scratch_storage_base
         }
         else if (__supports_USM_device)
         {
-            _T __tmp;
-            __q.memcpy(&__tmp, __scratch_buf.get() + __scratch_n + _Idx, 1 * sizeof(_T)).wait();
-            return __tmp;
+            static_assert(sycl::is_device_copyable_v<_T>,
+                          "The type _T must be device copyable to use __result_and_scratch_storage.");
+            // Avoid default constructor for _T, we know that _T is device copyable and therefore a copy construction
+            // is equivalent to a bitwise copy.  We may treat *__tmp as if it has been constructed.
+            oneapi::dpl::__internal::__lazy_ctor_storage<_T> __lazy_ctor_storage;
+            __q.memcpy(&(__lazy_ctor_storage.__v), __scratch_buf.get() + __scratch_n + _Idx, 1 * sizeof(_T)).wait();
+
+            // Setting up _T to be destroyed as this function exits. _T being device copyable provides that it has a
+            // public non deleted destructor.
```

Comment:
```suggestion
            // public non deleted destructor.
```
Destructor may be not public. But may it be really deleted?

Reply:
https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec::device.copyable
"Type T has a public non-deleted destructor;"

You can delete a destructor.  I'm not sure exactly the use case you would want this functionality for, but its possible.  
It stops the object from being stack allocated or deleted, or other situations which would result in a destructor being called.  Anyway, the reason I mention it is because the sycl spec mentions it.

Reply:
Have you tested it with different kinds of a destructor - trivial, non-trivial, deleted?